salt/states/group.py

```diff
@@ -299,10 +309,20 @@ def absent(name):
     if grp_info:
         # Group already exists. Remove the group.
         if __opts__['test']:
-            ret['result'] = None
-            ret['comment'] = 'Group {0} is set for removal'.format(name)
+            if onlyifempty and grp_info['members']:
+                ret['result'] = True
+                ret['comment'] = ('Group {0} will not be removed since members are '
+                                  'present and onlyifempty is set').format(name)
+            else:
+                ret['result'] = None
+                ret['comment'] = 'Group {0} is set for removal'.format(name)
             return ret
-        ret['result'] = __salt__['group.delete'](name)
+        if onlyifempty and grp_info['members']:
+            ret['comment'] = ('Group {0} will not be removed since members are '
```

Comment:
We shouldn't be using future tense here since the state will have completed. Instead we should say "was not removed".